cognite_toolkit/_cdf_tk/apps/_core_app.py

```diff
@@ -186,6 +187,14 @@ def build(
                 help="Do not check CDF for missing dependencies.",
             ),
         ] = False,
+        exit_on_warning: Annotated[
+            bool,
+            typer.Option(
+                "--exit-non-zero-on-warning",
+                "-w",
+                help="Exit with non-zero code on warning.",
+            ),
+        ] = False,
```

Comment:
You need to repeat the entire method with the new arg and use the flag to switch which one is selected.